### PR TITLE
tried to change the onChange in team search box

### PR DESCRIPTION
**Pull request by ItayNaumann**

### CODE REVIEW

**Comment by ItayNaumann** [comment]
got a bit long there, will add the firebase changes in another commit.

**Comment by itamarsch** [comment]
This PR looks Huge but it's mostly ui changes so git says it's a alot

File: lib/views/mobile/firebase_submit_button.dart [hunk 1]
```diff
@@ -124,7 +122,7 @@ class _FireBaseSubmitButtonState extends State<FireBaseSubmitButton> {
         running = false;
       } else if (event.state == TaskState.success) {
         final Map<String, dynamic> vars =
-            Map<String, dynamic>.from(widget.vars.toHasuraVars());
+            Map<String, dynamic>.from(widget.toHasuraVars());
```
Comment (itamarsch) [inline lib/views/mobile/firebase_submit_button.dart:125]: You don't need this call to `from`

File: lib/views/pc/team_info/team_info_screen.dart [hunk 2]
```diff
@@ -8,63 +8,62 @@ import "package:scouting_frontend/views/common/dashboard_scaffold.dart";
...
+              SizedBox(height: defaultPadding),
+              Expanded(
+                flex: 10,
+                child: team.mapNullable(
+                      TeamInfoData.new,
+                    ) ??
+                    NoTeamSelected(),
+              ),
+            ],
+          ),
```
Comment (itamarsch) [inline lib/views/pc/team_info/team_info_screen.dart:66]: Make this a function and have the function receive a body (in this case a `TeamInfo` or `NoTeamSelected`) and a searchBox
and then pass a call to the function with the correct parameters for buildWithTeam and buildWithoutTeam

**Review by itamarsch — CHANGES_REQUESTED** [review]

**Comment by giladwo** [comment]
image
image

There are way too many changes here, and they probably don't belong on the same PR.
Unless you can convince me they do, please split them into smaller PRs (shouldn't take long, provided the changes were 
carefully committed separately).

**Comment by giladwo** [comment]
We don't want a mega PR that handles all open issues. We want small PRs, even multiple ones per issue (when it makes sense, i.e. a methodological change that should be applied to different parts).

**Comment by ItayNaumann** [comment]
image
image
we are removing here not needed default values to send hasura,setting them as the default for the class and removing the to hasura vars function and moving it to the only time we use it and removing the team from the class. probably not needed in this pr should be on another pr.

**Comment by ItayNaumann** [comment]
image
switched to the new way we are using teamSelectionFuture

**Comment by ItayNaumann** [comment]
image
image
image
this is the main change of the pr, changing teamSearchBox, in here we are getting from where its called a couple of functions.
setTeam is to be able to have a onSelected cause we are setting up the widget here so the place it is called to be cant access it without it.
on build we have 2 functions; build with team and build without team. we are calling build with team when we have team and when its null we are calling build without team.
also using the setTeam function instead of using setState directly.

**Comment by ItayNaumann** [comment]
image
in here we probably went too far for this pr; removed the HasuraVars file and changed the submit button to get a toHasuraVars function

**Comment by ItayNaumann** [comment]
image
removed the toHasura vars in pit vars and removed the team from pit vars

**Comment by ItayNaumann** [comment]
image
in here (fault view) its just changed to the new way we use teamSelectionFuture and we removed team from it

**Comment by ItayNaumann** [comment]
in input view the git diff is no better than the normal diff, but we just removed lightTeam from there, changed to the new way it shows only the match and team search boxes, but then it builds the whole screen after he gets both of these variables,also defined the parsing "toHasuraVars" in the file instead of in the vars of  this file

**Comment by ItayNaumann** [comment]
image
changed to new teamSelectionFuture, changed from a list of controllers to each controller by himself and moved the parse to hasura vars to here

**Comment by ItayNaumann** [comment]
in specific view its pretty much the same as input view; removed the list of controllers, changed to new teamSelectionFuture version and moved the definition of the toHasuraVars to here and changed that the screen wouldnt be shown until the user inputed a match and a team

**Comment by ItayNaumann** [comment]
side nav bar just changed to new team selecitonFuture
image

**Comment by ItayNaumann** [comment]
image
in specific vars simply removed the toHasuraVars function and removed lightTeam from being defined there

**Comment by ItayNaumann** [comment]
image
in the submit button just changed it from getting a map of string to dynamic to a function that returns that map

**Comment by ItayNaumann** [comment]
image
in teamSelectionMatches just changed to include new features from teamSelectionFuture

**Comment by ItayNaumann** [comment]
image
in compare screen we remove all of the controllers from this scope and add before the return of the search box the teams that arent in the list full of teams

**Comment by ItayNaumann** [comment]
image
image
image
making an empty array of void functions that reset the search box and defining each one of them to a different search box and then after we send the info to the db we reset all of them at once

**Comment by ItayNaumann** [comment]
image
removing the sending to db here...

**Comment by ItayNaumann** [comment]
image
changing to new way of teamSelectionFuture

**Comment by ItayNaumann** [comment]
I think i can make this pr 2 pr, one for removing the hasura things and one for all the others

**Comment by amitcha06** [comment]
> image in here we probably went too far for this pr; removed the HasuraVars file and changed the submit button to get a toHasuraVars function

Yeah These should NOT be on the same PR

**Comment by amitcha06** [comment]
> in input view the git diff is no better than the normal diff, but we just removed lightTeam from there, changed to the new way it shows only the match and team search boxes, but then it builds the whole screen after he gets both of these variables,also defined the parsing "toHasuraVars" in the file instead of in the vars of this file

Note that #356 is not merged yet, so there will be merge conflicts

**Comment by amitcha06** [comment]
> in specific view its pretty much the same as input view; removed the list of controllers, changed to new teamSelectionFuture version and moved the definition of the toHasuraVars to here and changed that the screen wouldnt be shown until the user inputed a match and a team

i would argue that anything unrelated to TeamSelection should not be in this PR (the controllers, for example)

**Comment by amitcha06** [comment]
> image in specific vars simply removed the toHasuraVars function and removed lightTeam from being defined there

Same as stated above, the HasuraVars shouldnt be in this PR

**Comment by amitcha06** [comment]
> image in the submit button just changed it from getting a map of string to dynamic to a function that returns that map

^

**Comment by amitcha06** [comment]
> image image image image making an empty array of void functions that reset the search box and defining each one of them to a different search box and then after we send the info to the db we reset all of them at once and defining the sending to db here..

These should also be split

**Comment by amitcha06** [comment]
> image removing the sending to db here...

^

**Comment by amitcha06** [comment]
> I think i can make this pr 2 pr, one for removing the hasura things and one for all the others

They can probably even more. 
one for the TeamSelection
one for HasuraVars
one for the changes in matches_screen,
etc...

**Pull request closed without merging**